stellargraph/core/convert.py

```diff
@@ -113,53 +182,68 @@ def convert(self, elements):
             type_name: self._convert_single(type_name, data)
             for type_name, data in elements.items()
         }
-        return (
-            {type_name: shared for type_name, (shared, _) in singles.items()},
-            {type_name: features for type_name, (_, features) in singles.items()},
-        )
+
+        shared, type_starts = self._shared_and_starts_from_singles(singles)
+        features = {type_name: features for type_name, (_, features) in singles.items()}
+        return (shared, type_starts, features)
 
 
 def convert_nodes(data, *, name, default_type, dtype) -> NodeData:
     converter = ColumnarConverter(
         name,
         default_type,
+        type_column=None,
```

Comment:
I considered this in #1183:

> The same can apply to nodes, although is potentially less common, since different node types may have different feature dimensions, and, different node types can also just be encoded as categorical features.

In particular, I've not seen knowledge graphs with different node types, but rather each node is an entity all by itself. This is visible in `ComplEx`/`DistMult` where the node feature input into the bilinear product is just looked-up in an embedding layer, and in `RGCN`, where the nodes must all have a single type/the same feature dimension.

I didn't do it here, because I didn't want to spend the effort teasing out the right abstraction for catching/handling features. I'll open a follow up issue.

Reply:
fair point, lets at it in if/when we come across a graph/algorithm that needs it 👍